### PR TITLE
MOBT-801: Wet symbol cloud contributions to dry summary determination

### PR DESCRIPTION
**Pull request by bayliffe**
Address issue of discounting cloud associated with wet symbols when determining summary symbol. If we have a day containing a mix of clear, cloudy, and wet conditions, the cloud associated with the wet periods is not currently considered if overall the day is dry. That means that we are over forecasting sunny / clear conditions.

This change ensures that the dry symbol determination, for locations that are destined to receive a dry summary symbol, includes the wet symbols nearest cloud equivalent.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

File: improver/wxcode/modal_code.py [hunk 1]
```diff
@@ -733,6 +739,35 @@ def _set_blended_times(cube: Cube, result: Cube) -> None:
             )
             result.replace_coord(new_coord)
 
+    def _get_cloud_equivalents(
```
Comment (brhooper) [inline improver/wxcode/modal_code.py:742]: Would the purpose of this function be clearer if it were called `get_dry_equivalents` or `get_dry_cloud_equivalents`?

**Review by brhooper — COMMENTED** [review]
Thanks @bayliffe, I've added a couple of comments.

File: improver_tests/wxcode/wxcode/test_ModalFromGroupings.py [hunk 2]
```diff
@@ -61,6 +61,8 @@
     "snow_shower": [26, 23],
     "snow": [27, 24],
     "thunder": [30, 29],
+    "cloud": [7, 8],
```
[the remark this answers is not in the document]
Comment (bayliffe) [reply improver_tests/wxcode/wxcode/test_ModalFromGroupings.py:64]: My thinking was that we are already significantly increasing the instances where we will get cloud. By hedging this to simply cloudy for tied situations I leave a glimmer of optimism. My genuine thoughts when deciding this.

Will you take this optimism from the forecast?

[the remark this answers is not in the document]
Comment (brhooper) [reply improver_tests/wxcode/wxcode/test_ModalFromGroupings.py:64]: We've decided to maintain the existing priority ordering within the cloud grouping.

**Review by brhooper — APPROVED** [review]
Thanks @bayliffe, this all looks fine to me.

I'm impressed that you managed to fix a typo before I had even commented on it - that's good service.

**Review by Kat-90 — APPROVED** [review]

**Pull request merged**